iaso/models/org_unit.py

```diff
@@ -100,12 +100,17 @@ def filter_for_user_and_app_id(
         if app_id is not None:
             try:
                 project = Project.objects.get_for_user_and_app_id(user, app_id)
-                queryset = queryset.filter(projects__in=[project])
+                queryset = self.filter_for_project(project, queryset)
             except Project.DoesNotExist:
                 return self.none()
 
         return queryset
 
+    def filter_for_project(self, project: Project, queryset=None):
+        if queryset is None:
+            queryset = self.all()
```

Comment:
I'm not sure if this is actually going to change anything performance-wise, but it should be better
```suggestion
            queryset = self
```